staging/src/k8s.io/client-go/util/certificate/certificate_manager.go

```diff
@@ -233,6 +244,25 @@ func NewManager(config *Config) (Manager, error) {
 		now:                     time.Now,
 	}
 
+	name := config.Name
+	if len(name) == 0 {
+		name = m.signerName
+	}
+	if len(name) == 0 {
+		switch {
+		case hasKeyUsage(config.Usages, certificates.UsageClientAuth):
+			name = string(certificates.UsageClientAuth)
+		default:
+			name = "certificate"
+		}
+	}
+
+	m.name = name
+	m.logf = config.Logf
+	if m.logf == nil {
+		m.logf = func(format string, args ...interface{}) { klog.V(2).Infof(format, args...) }
+	}
```

Comment:
As a small non-blocking nit I'd suggest not doing this because it'll likely need to get ripped out when we migrate this to structured logs.

Reply:
We can't just slam structured logs into client-go - it's a library, not a full featured function.  Or are you saying that everyone who uses client-go is required to use klog (which is something historically we've said we wouldn't do)

Reply:
Oh, right, this is client-go and you're correct that we're not doing that; morning brain fog :)

Reply:
it’s ok, i wasn’t positive it wasn’t me that missed something :)